ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1215,30 +1223,128 @@ private String checkOtherDrawsQuantity(String id, Date date, String rowObjectId,
+            StringBuilder errorMsgBuilder = new StringBuilder();
+
+            //always report the most severe overage
+            if (!overages.isEmpty())
+            {
+                errorMsgBuilder.append("Blood volume of ")
+                               .append(rowQuantity)
+                               .append(" (")
+                               .append(overages.descendingSet().iterator().next())
+                               .append(" over ")
+                               .append(interval)
+                               .append(" days) exceeds the allowable volume of ")
+                               .append(maxAllowable)
+                               .append(" mL (weight: ")
+                               .append(weight)
+                               .append(" kg).\n");
+            }
+
+            //report all the ones that are close to limit
+            for (double closeValue : closeToThreshold.descendingSet())
+            {
+                errorMsgBuilder.append("Limit notice! Blood volume of ")
+                               .append(rowQuantity)
+                               .append(" (")
+                               .append(closeValue)
+                               .append(" over ")
+                               .append(interval)
+                               .append(" days) is within ")
+                               .append(getBloodNearingOveragesThreshold())
+                               .append(" mL of the max allowable limit of ")
+                               .append(maxAllowable)
+                               .append(" mL (weight: ")
+                               .append(weight)
+                               .append(" kg).\n");
+            }
+
+            return errorMsgBuilder.toString();
+        }
+        else
+        {
+            return null; // No errors
         }
+    }
 
-        return null;
+    /**
+     * Gets the center specific threshold to warn when blood vols are close to the max blood allowed,
+     * only used if the doWarnForBloodNearOverages() is true
+     * e.g., if the max allowable blood drawn vol is 60.0, a threshold of 4.0 will warn users if blood vol is greater than 56.0 ml
+     * this should be set on the JS side with helper.setCenterCustomProps()
+     *
+     * @return      the threshold value of the limit
+     */
+    public double getBloodNearingOveragesThreshold()
+    {
+        Object theVal = _centerCustomProps.get("bloodNearOverageThreshold");
+        if (null != theVal)
+        {
+            if (theVal instanceof Integer)
+            {
+                Integer theValInt = (Integer) theVal;
+                return theValInt.doubleValue();
+            }
+            else if (theVal instanceof Double)
+            {
+                return (Double) theVal;
+            }
```

Comment:
can you please either throw or log an error message if theVal is neither an integer or double? otherwise it will just get silently ignored.

Reply:
Thanks. Two things:

1) I think if the module supplies a value, but that value is invalid, it should be error(), not warn()
2) The second warn() right below that doesnt seem right to me. If we except the developer to provide a threshold, we should just enforce that and not default to anything. If we think it's valid for the develop to just accept the default, then I dont think we should warn() about it - that's just noise. Maybe there is an argument for debug() logging.